Source/DataSources/GpxDataSource.js

```diff
@@ -0,0 +1,1065 @@
+        PolylineGraphics,
+        PolylineOutlineMaterialProperty,
+        PositionPropertyArray,
+        SampledPositionProperty,
+        TimeIntervalCollectionProperty) {
+    "use strict";
+
+    var parser = new DOMParser();
+    var autolinker = new Autolinker({
+        stripPrefix : false,
+        twitter : false,
+        email : false,
+        replaceFn : function(linker, match) {
+            if (!match.protocolUrlMatch) {
+                //Prevent matching of non-explicit urls.
+                //i.e. foo.id won't match but http://foo.id will
+                return false;
+            }
+        }
+    });
+
+    var BILLBOARD_SIZE = 32;
+
+    function readBlobAsText(blob) {
+        var deferred = when.defer();
+        var reader = new FileReader();
+        reader.addEventListener('load', function() {
+            deferred.resolve(reader.result);
+        });
+        reader.addEventListener('error', function() {
+            deferred.reject(reader.error);
+        });
+        reader.readAsText(blob);
+        return deferred;
+    }
+
+    function proxyUrl(url, proxy) {
+        if (defined(proxy)) {
+            if (new Uri(url).isAbsolute()) {
+                url = proxy.getURL(url);
+            }
+        }
+        return url;
+    }
+
+    function getOrCreateEntity(node, entityCollection) {
+        var id = queryStringAttribute(node, 'id');
+        id = defined(id) ? id : createGuid();
+        var entity = entityCollection.getOrCreateEntity(id);
+        return entity;
+    }
+
+    function readCoordinate(value) {
+        if (!defined(value)) {
+            return undefined;
+        }
+
+        var digits = value.match(/[^\s,\n]+/g);
+        if (digits.length !== 2 && digits.length !== 3) {
+            window.console.log('GPX - Invalid coordinates: ' + value);
+            return undefined;
+        }
+
+        var longitude = parseFloat(digits[0]);
+        var latitude = parseFloat(digits[1]);
+        var height = parseFloat(digits[2]);
+
+        longitude = isNaN(longitude) ? 0.0 : longitude;
+        latitude = isNaN(latitude) ? 0.0 : latitude;
+        height = isNaN(height) ? 0.0 : height;
+
+        return Cartesian3.fromDegrees(longitude, latitude, height);
+    }
+
+    function readCoordinates(element) {
+        if (!defined(element)) {
+            return undefined;
+        }
+
+        var tuples = element.textContent.match(/[^\s\n]+/g);
+        var length = tuples.length;
+        var result = new Array(length);
+        var resultIndex = 0;
+        for (var i = 0; i < length; i++) {
+            result[resultIndex++] = readCoordinate(tuples[i]);
+        }
+        return result;
+    }
+
+    function readCoordinateFromNode(node) {
+        var longitude = queryNumericAttribute(node, 'lon');
+        var latitude = queryNumericAttribute(node, 'lat');
+        var elevation = queryNumericValue(node, 'ele', namespaces.gpx);
+        elevation = 0; //TODO disregard elevation for now
```

Comment:
That problem will go away very soon, so let's not worry about it for now.  Lots of Cesium users use terrain anyway and lots of people have GPX data from aerial vehicles (like drones), so allowing height data is important.